src/faction_camp.cpp

```diff
@@ -428,18 +428,22 @@ static bool update_time_fixed( std::string &entry, const comp_list &npc_list,
     return avail;
 }
 
-static basecamp *get_basecamp( npc &p )
+static cata::optional<basecamp *> get_basecamp( npc &p )
 {
-    basecamp *bcp = g->m.camp_at( p.pos(), 60 );
+
+    tripoint omt_tri = p.global_omt_location();
+    cata::optional<basecamp *> bcp = overmap_buffer.find_camp( omt_tri.x, omt_tri.y );
     if( bcp ) {
         return bcp;
     }
     g->m.add_camp( p.pos(), "faction_camp" );
-    bcp = g->m.camp_at( p.pos(), 60 );
-    if( bcp ) {
-        bcp->define_camp( p );
+    bcp = overmap_buffer.find_camp( omt_tri.x, omt_tri.y );
+    if( !bcp ) {
+        return cata::nullopt;
     }
-    return bcp;
+    basecamp *temp_camp = *bcp;
```

Comment:
why aren't you calling define_camp() on bcp and returning bcp?

Reply:
Because it doesn't work, something to do with it being a cataoptional.
Can't run define_camp on it directly, have to create a basecamp pointer to it.
I'm not entirely sure why

Reply:
Okay.